File: pre-contest/b_jeu_des_batonnets/solutions/b_jeu_des_batonnets.cs
using System;

namespace c_sharp
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            string[] s = Console.ReadLine().Split(' ');
            var nb = int.Parse(s[0]);
            var b = int.Parse(s[1]);
            var j = int.Parse(Console.ReadLine());
            var reste = nb % (b + 1);

            if ((j == 0 && (reste != 0)) || (j == 1 && (reste == 0)))
            {
                Console.WriteLine("OUI");
            }
            else
            {
                Console.WriteLine("NON");
            }
        }
    }
}

File: pre-contest/b_jeu_des_batonnets/solutions/b_jeu_des_batonnets.java
import java.util.Scanner;

public class Nim2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int nb = sc.nextInt();
        int b = sc.nextInt();
        int j = sc.nextInt();
        int reste = nb%(b+1) ;
        if(( j==0 && ( reste !=0 ) ) || (j==1 && ( reste==0 ))){
            System.out.println("OUI");
        }else {
            System.out.println("NON");
        }

    }
}
